### PR TITLE
Kotlin 1.9.20 & refreshVersions plugin

### PR DESCRIPTION
**Pull request by jillesvangurp**
- Updated to kotlin 1.9.0 and related fixes to make that work

- Upgraded build to use the refreshVersions plugin (see https://splitties.github.io/refreshVersions/)

Reason for refreshVersions is that manually checking all the versions is a bit of a pain. And I've done this a few times now on this project.

Note, refresh versions can also use the toml format that gradle has these days. Not a big fan myself but should be easy to switch. All the npms, gradle plugins, and jar versions are managed currently, which is nice. Anything that can't be updated to latest is documented with a comment in versions.properties.

### CODE REVIEW

**Comment by jillesvangurp** [comment]
Added some additional changes that update to kotlin 1.9.10 (ksp plugin was updated)

Also updates gradle to the latest now

**Comment by jillesvangurp** [comment]
Understood. I've been using refreshVersions on all my projects for some time now. Works great and saves a lot of time. 

Let's re-rerun `gradle refreshVersions` before merging this and make sure everything is up to date.

**Comment by Lysander** [comment]
Please be patient some more week. I think at the beginning of October we can merge this PR - definitely looking forward to it! 👍

**Comment by jillesvangurp** [comment]
No worries, I pushed this to our own repository so it's not blocking me.

I built https://github.com/jillesvangurp/rankquest-studio with it a few weeks ago.

**Comment by jillesvangurp** [comment]
No worries, I pushed this to our own repository so it's not blocking me.

I built https://github.com/jillesvangurp/rankquest-studio with it a few weeks ago. I actually used the k2 compiler with that, which is a massive enabler for faster build speeds

I just pushed a few more updates to the branch. Mostly some patch releases for the NPMs

**Comment by Lysander** [comment]
I am looking forward to it! I am very curious, how the K2 compiler improves the build experience.

**Comment by jillesvangurp** [comment]
I just applied the changes from RC-12. So, everything is synced up again.

**Comment by jillesvangurp** [comment]
Updated to 1.9.20

**Review by jamowei — DISMISSED** [review]
Thank you very much for your effort!

**Review by jamowei — APPROVED** [review]

File: publishing.gradle.kts [hunk 1]
```diff
@@ -12,15 +12,15 @@ fun Project.publishing(action: PublishingExtension.() -> Unit) =
     configure(action)
 
 
-signing {
+/*signing {
```
[the remark this answers is not in the document]
Comment (jillesvangurp) [reply publishing.gradle.kts]: sure, I think that was accidental when I published to our own repo

File: serialization/build.gradle.kts [hunk 2]
```diff
@@ -5,12 +5,12 @@ plugins {
 
 kotlin {
     jvm()
-    js(BOTH).browser { }
+    js().browser { }
```
[the remark this answers is not in the document]
Comment (jillesvangurp) [reply serialization/build.gradle.kts]: BOTH is deprecated

**Pull request merged**